src/main/java/org/web3j/codegen/SolidityFunctionWrapperGenerator.java

```diff
@@ -432,37 +439,99 @@ private static void buildTransactionFunction(
+        TypeSpec.Builder builder = TypeSpec.classBuilder(className).addModifiers(Modifier.PUBLIC, Modifier.STATIC);
+        for (NamedTypeName namedType : indexedParameters) {
+            builder.addField(namedType.getTypeName(), namedType.getName(), Modifier.PUBLIC);
+        }
 
-        String functionName = functionDefinition.getName();
-        String generatedFunctionName =
-                "process" + Strings.capitaliseFirstLetter(functionName) + "Event";
+        for (NamedTypeName namedType : nonIndexedParameters) {
+            builder.addField(namedType.getTypeName(), namedType.getName(), Modifier.PUBLIC);
+        }
+
+        return builder.build();
+    }
+
+    static MethodSpec buildEventObservableFunction(String responseClassName, String functionName, List<NamedTypeName> indexedParameters, List<NamedTypeName> nonIndexedParameters) throws ClassNotFoundException {
+        String generatedFunctionName = functionName + "EventObservable";
+        ParameterizedTypeName parameterizedTypeName = ParameterizedTypeName.get(ClassName.get(rx.Observable.class), ClassName.get("", responseClassName));
+
+        MethodSpec.Builder observableMethodBuilder = MethodSpec.methodBuilder(generatedFunctionName)
+                .addModifiers(Modifier.PUBLIC)
+                .returns(parameterizedTypeName);
+
+
+        buildVariableLengthEventConstructor(
+                observableMethodBuilder, functionName, indexedParameters, nonIndexedParameters);
+
+        observableMethodBuilder.addStatement("$1T filter = new $1T($2T.EARLIEST,$2T.LATEST, getContractAddress())", EthFilter.class, DefaultBlockParameterName.class)
+                .addStatement("filter.addSingleTopic($T.encode(event))", EventEncoder.class)
+                .beginControlFlow("return web3j.ethLogObservable(filter).map(log ->")
+                .addStatement("$T eventValues = extractEventParameters(event, log)", EventValues.class)
+                .addStatement("$1T typedResponse = new $1T()", ClassName.get("", responseClassName))
+                .addCode(buildTypedResponse("typedResponse", indexedParameters, nonIndexedParameters))
+                .addStatement("return typedResponse")
+                .endControlFlow(")");
+
+        return observableMethodBuilder
+                .build();
+    }
 
-        MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(generatedFunctionName)
+    static MethodSpec buildEventTransactionReceiptFunction(String responseClassName, String functionName, List<NamedTypeName> indexedParameters, List<NamedTypeName> nonIndexedParameters) throws ClassNotFoundException {
+        ParameterizedTypeName parameterizedTypeName = ParameterizedTypeName.get(ClassName.get(List.class), ClassName.get("", responseClassName));
+
+        String generatedFunctionName = "get" + Strings.capitaliseFirstLetter(functionName) + "EventFromTransactionReceipt";
```

Comment:
You don't need to include "FromTransactionReceipt" in the generated method name, as the TransactionReceipt is an explicit parameter.